pkg/services/ngalert/schedule/schedule.go

```diff
@@ -305,7 +306,8 @@ func (sch *schedule) processTick(ctx context.Context, dispatcherGroup *errgroup.
 	return readyToRun, registeredDefinitions
 }
 
-func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertRuleKey, evalCh <-chan *evaluation, updateCh <-chan ruleVersion) error {
+func (sch *schedule) ruleRoutine(grafanaCtx context.Context, rule *ngmodels.AlertRule, evalCh <-chan *evaluation, updateCh <-chan ruleVersion) error {
+	key := rule.GetKey()
```

Comment:
I would recommend reverting this change as it makes very unclear which version of the rule is being used. The rule is provided as part of eval command, which contains only the most recent version. Then, based on the diff between local copy and one from command, this method does some state management.